cmd/juju/commands/debuglog.go

```diff
@@ -213,36 +213,49 @@ func (c *debugLogCommand) processEntities(isCAAS bool, entities []string) []stri
 	if entities == nil {
 		return nil
 	}
-	result := make([]string, len(entities))
+	results := make([]string, len(entities))
 	for i, entity := range entities {
-		// A stringified unit or machine tag never match their "IsValid"
+		// Filter out valid tags early
+		if tag, err := names.ParseTag(entity); err == nil {
+			if tag.Kind() == names.UnitTagKind || tag.Kind() == names.MachineTagKind || tag.Kind() == names.ApplicationTagKind {
+				results[i] = entity
+				continue
+			}
+			logger.Warningf("Tag %s of kind %s is not a supported filter for debug-log", entity, tag.Kind())
```

Comment:
We need to reword this to make it more user friendly. Most users have no idea what the juju tags are. 

Something like?
"WARNING charm-juju-qa-test is not a valid type for use with the --include flag. Expected examples include 0, machine-0, ubuntu, application-ubuntu, ubuntu/0"

Same for the other logger messages

Reply:
Agreed. Have re-written